sim/dex.ts

```diff
@@ -570,6 +570,35 @@ dexes['base'] = new ModdedDex();
 dexes[BASE_MOD] = dexes['base'];
 
 export const Dex = dexes['base'];
+
+// Populate _toIDCache with data from the base mod.
+// It's representative enough for all other mods, so we don't need to dynamically grow the cache.
+{
+	const cache = Data._toIDCache;
+	const doName = (name?: string) => {
+		if (!name) return;
+		const id = toID(name);
+		const old = cache.get(name);
+		if (old === undefined) cache.set(name, id);
+		else if (old !== id) throw new Error("internal error with ID caching logic");
```

Comment:
No need for all these defensive statements.

Just ``if (!cache.has(name)) cache.set(name, id)``. Which makes the entire function short enough that it only makes sense to inline it.

Reply:
My thinking with checking `(old !== id)` is that it would possibly catch faulty cache init logic elsewhere.
I guess the documentation covers that.
Will do.